yileifeng/ramp4-pcar4
# fix grid fixture

I had the best time with this. Trust me. 🗑️ 🔥 

---
**Changes in this PR**
- updated ag-grid-vue from `v25.3.0` -> `v26.0.0`
- grid fixture now opens and functions
- updated grid + all custom grid components to use defineComponent
- `floatingFilters` property no longer exists on gridOptions (deprecated), so I had to modify the `toggle filters` button to individually update `floatingFilters` on each columnDef.

**Concerns**
- when using the column visibility dropdown, clicking an option will correctly hide/show a column, however the checkmark is not updated. (update: this seems to work in the demo, not locally)
- Custom filter style isn't working
- I had to temporarily remove vue-tippy from the grid sub-components (custom filters, zoom/details buttons, etc.) in order for this to work
- Keyboard accessibility is not working the same as it is currently working on the master branch.
- when you open the grid, the console is spammed with a warning saying "[Vue warn]: Data property "params" is already defined in Props.". This is probably related to how ag-grid passes in a `params` prop to its components, but I'm unable to find anything online about this and can't find a workaround.

**Testing this PR**
You can find a demo for this PR [here](http://ramp4-app.azureedge.net/demo/users/RyanCoulsonCA/vue3-grid-fix/host/index.html).

To test, open the grid for each of the layers and ensure the following things work:
- if the layer is not visible on the map, then the grid should be empty
- make sure filtering works for each filter type
- make sure details/zoom buttons work
- make sure global filter works
- make sure the `apply filters to map` button works
- make sure show/hide filters works
- test out the `clear filters` option in the dropdown menu at the top
- make sure re-arranging columns works
- make sure the `visible rows` text updates correctly when you filter and scroll the grid
- do anything else you can think of to break it
